v2/internal/pkg/helm/local_stored_collector.go

```diff
@@ -295,7 +295,7 @@ func createIndexFile(indexURL string) (helmrepo.IndexFile, error) {
 
 	indexFile, err := parser.ParseYamlReader[helmrepo.IndexFile](resp.Body)
 	if err != nil {
-		return helmrepo.IndexFile{}, err
+		return helmrepo.IndexFile{}, fmt.Errorf("%w for %s", err, indexURL)
```

Comment:
Can we add more helpful context? Maybe
```suggestion
		return helmrepo.IndexFile{}, fmt.Errorf("failed to parse %q into index file: %w", indexURL, err)
```